File: Extra/274-H-Index.py
#
# Created by egoistpizza on 8.10.2025.
#

# 274. H-Index
# https://leetcode.com/problems/h-index/

from typing import List

class Solution:
    def hIndex(self, citations: List[int]) -> int:

        n = len(citations)
        count = [0] * (n + 1)

        for c in citations:
            count[min(c, n)] += 1

        papers = 0

        for h in range(n, -1, -1):
            papers += count[h]

            if papers >= h:
                return h

        return 0

File: 07-10-2025/274-H-Index .cpp
//
// Created by egoistpizza on 7.10.2025.
//

// 274. H-Index
// https://leetcode.com/problems/h-index/

#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int hIndex(vector<int>& citations) {
        int n = citations.size();
        vector<int> count(n+1,0);

        for (int c : citations){
            if (c >= n) count[n]++;
            else count[c]++;
        }

        int papers = 0;
        for (int h = n; h >= 0; h--){
            papers += count[h];
            if (papers >= h){
                return h;
            }
        }
        return 0;
    }
};
